File: Data Structure/Python/Queue.py
class Queue:
    def __init__(self):
        self.queue = ()
        self.front = -1
        self.back = -1

    def getFront(self):
        return self.front

    def getBack(self):
        return self.back

    def Queue(self):
        return self.Queue()

    def offer(self, item):
        self.Queue().add(item)
        if self.getFront() == -1 and not (self.Queue().get(0) is None):
            self.front = 0
        self.back = self.back + 1
        return self.Queue().get(self.getBack())

    def poll(self):
        backup = self.Queue().get(self.getFront())
        self.front = self.front + 1
        return backup

    def peek(self):
        return self.Queue().get(self.getFront())

    def display(self):
        print("\n")
        string = "List: "
        for location in range(self.getFront(), self.getBack()+1):
            if location == self.getBack():
                string = string + self.Queue().get(location)
            else:
                string = string + self.Queue().get(location) + ", "
        print(string, "\n")

File: Data Structure/Flutter/Dart/queue.dart
class Queue<E> {
  List<E> queueArray = [];
  int size = 0, front = 0, rear = -1;

  Queue();

  int getFront(){
    return front;
  }
  int getRear(){
    return rear;
  }
  int getSize(){
    return size;
  }
  List<E> getQueueArray(){
    return queueArray;
  }
  void offer(E item){
    rear = getRear()+1;
    queueArray[getRear()] = item;
    size = getSize()+1;
  }
  E poll(){
    front = getFront()+1;
    return queueArray[getFront()-1];
  }
  E element(){
    return queueArray[getFront()];
  }
  void display(){
    print("Size = ${getSize()} \n");
    print("Queue: [");
    getQueueArray().forEach((action){
      print("$action, ");
    });
    print("]");
  }
}